common/djangoapps/track/shim.py

```diff
@@ -113,6 +117,9 @@ def __call__(self, event):
             return
         payload = event['event']
 
+        if isinstance(payload, basestring):
+            return
```

Comment:
I'm afraid I still don't understand this.  Processors run in order, and the legacy shim should be applied before the video shim.  The legacy shim converts the data field into an event field, and makes sure that it's a string instead of a dict.  So I would expect the video payload here to always be a string (or an empty dict).